Multisig/UI/Safe Management/Create Safe/Setup Safe/SelectAddressViewController.swift

```diff
@@ -0,0 +1,192 @@
+//
+//  SelectAddressViewController.swift
+//  Multisig
+//
+//  Created by Dmitry Bespalov on 28.02.22.
+//  Copyright © 2022 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import Web3
+
+class SelectAddressViewController: UIAlertController {
+    var completion: (Address) -> Void = { _ in }
+    var chain: Chain?
+    weak var presenter: UIViewController?
+
+    convenience init(chain: Chain?, presenter: UIViewController, completion: @escaping (Address) -> Void) {
+        self.init(title: nil, message: nil, preferredStyle: .actionSheet)
+        self.presenter = presenter
+        self.chain = chain
+        self.completion = completion
+
+        addAction(UIAlertAction(title: "Paste from Clipboard", style: .default, handler: handlePaste(_:)))
+        addAction(UIAlertAction(title: "Scan QR Code", style: .default, handler: handleScan(_:)))
+        addAction(UIAlertAction(title: "Address Book", style: .default, handler: handleAddressBook(_:)))
+        addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: nil))
+    }
+
+    func handlePaste(_ action: UIAlertAction) {
+        let text = Pasteboard.string
+        handle(text: text)
+    }
+
+    func handleScan(_ action: UIAlertAction) {
+        let vc = QRCodeScannerViewController()
+        vc.scannedValueValidator = { [unowned self] value in
+            do {
+                return try .success(validate(text: value).checksummed)
+            } catch {
+                return .failure(error)
+            }
+        }
+        vc.modalPresentationStyle = .overFullScreen
+        vc.delegate = self
+        vc.setup()
+        presenter?.present(vc, animated: true, completion: nil)
+    }
+
+    func handleAddressBook(_ action: UIAlertAction) {
+        let addressBookVC = AddressBookListTableViewController()
+        addressBookVC.filterByChain = chain
+        addressBookVC.isPickerModeEnabled = true
+        addressBookVC.onSelect = { [unowned addressBookVC, unowned self] address in
+            addressBookVC.dismiss(animated: true) {
+                completion(address)
+            }
+        }
+        let vc = ViewControllerFactory.modal(viewController: addressBookVC)
+        presenter?.present(vc, animated: true)
+    }
+
+    func validate(text: String?) throws -> Address {
```

Comment:
In Address.swift, we have method called addressWithPrefix which do part of the logic here, can we adapt it so we have the full logic all over the app ?

Reply:
I think this would be good, but I don't want to spend more time on that right now. This would require refactoring/redoing other code in other places.